src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp

```diff
@@ -0,0 +1,741 @@
+    perf_print_counter(_comms_errors);
+    perf_print_counter(_sample_perf);
+}
+
+int VL53L1X::probe()
+{
+    uint8_t ret, data;
+    ret = VL53L1_RdByte(1, &data);
+    if ((ret != PX4_OK) | (data != _device_id.devid_s.address)) {
+        return -EIO;
+    }
+    return PX4_OK;
+}
+
+void VL53L1X::RunImpl()
+{
+    uint8_t dataReady;
+
+    VL53L1X_CheckForDataReady(&dataReady);
+    if (dataReady == 1) {
+        collect();
+    }
+
+    ScheduleDelayed(VL53L1X_SAMPLE_RATE);
+}
+
+void VL53L1X::start()
+{
+    // Schedule the first cycle.
+    ScheduleNow();
+}
+
+int VL53L1X::init() {
+    int ret = PX4_OK;
+
+    ret = device::I2C::init();
+    if (ret != PX4_OK) {
+        return PX4_ERROR;
+    }
+
+    ret |= VL53L1X_SensorInit();
+    ret |= VL53L1X_ConfigBig(2, VL53L1X_SAMPLE_RATE);
+    ret |= VL53L1X_SetInterMeasurementInMs(VL53L1X_SAMPLE_RATE);
+    ret |= VL53L1X_StartRanging();
+    if (ret != PX4_OK) {
+        return PX4_ERROR;
+    }
+
+    PX4_INFO("vl53l1x init success");
+    return PX4_OK;
+}
+
+void VL53L1X::stop(){
+    VL53L1X_StopRanging();
+}
+
+void VL53L1X::print_usage()
+{
+    PRINT_MODULE_USAGE_NAME("vl53l1x", "driver");
+    PRINT_MODULE_USAGE_SUBCATEGORY("distance_sensor");
+    PRINT_MODULE_USAGE_COMMAND("start");
+    PRINT_MODULE_USAGE_PARAMS_I2C_SPI_DRIVER(true, false);
+    PRINT_MODULE_USAGE_PARAM_INT('R', 25, 1, 25, "Sensor rotation - downward facing by default", true);
+    PRINT_MODULE_USAGE_DEFAULT_COMMANDS();
+}
+
+/* ST */
+// int8_t VL53L1X::VL53L1_WaitMs(int32_t wait_ms){
+//     ScheduleDelayed(wait_ms * 1000); // us
+//     return 0;
+// }
+
+int8_t VL53L1X::VL53L1_RdByte(uint16_t index, uint8_t *data) {
+    int8_t ret;
+    uint8_t data_local[2];
+
+    // Convert
+    data_local[0] = (index >> 8) & 0xff;
+    data_local[1] = index & 0xff;
+
+    // Write register address to the sensor.
+    ret = transfer(&data_local[0], sizeof(data_local), nullptr, 0);
+    if (ret != PX4_OK) {
+        perf_count(_comms_errors);
+        return ret;
+    }
+
+    // Read from the sensor.
+    ret = transfer(nullptr, 0, data, 1);
+    if (ret != PX4_OK) {
+        perf_count(_comms_errors);
+        return ret;
+    }
+
+    return PX4_OK;
+}
+
+int8_t VL53L1X::VL53L1_RdWord(uint16_t index, uint16_t *data) {
+    int8_t ret;
+    uint8_t data_local[2], data_out[2];
+
+    // Convert
+    data_local[0] = (index >> 8) & 0xff;
+    data_local[1] = index & 0xff;
+
+    // Write register address to the sensor.
+    ret = transfer(&data_local[0], sizeof(data_local), nullptr, 0);
+    if (ret != PX4_OK) {
+        perf_count(_comms_errors);
+        return ret;
+    }
+
+    // Read from the sensor.
+    ret = transfer(nullptr, 0, data_out, 2);
+    if (ret != PX4_OK) {
+        perf_count(_comms_errors);
+        return ret;
+    }
+
+    *data = ((uint16_t)data_out[0]<<8) + (uint16_t)data_out[1];
+
+    return PX4_OK;
+}
+
+// int8_t VL53L1X::VL53L1_RdDWord(uint16_t index, uint32_t *data) {
```

Comment:
Are these going to be needed later? If not, please delete all the dead commented code.

Reply:
I thought it might be useful for the ability to add some functions from the original driver